designer/server/src/main/scala/pl/touk/nussknacker/ui/process/processingtype/loader/ProcessingTypesConfigBasedProcessingTypeDataLoader.scala

```diff
@@ -6,64 +6,76 @@ import pl.touk.nussknacker.engine._
       getModelDependencies: ProcessingType => ModelDependencies,
       getDeploymentManagerDependencies: ProcessingType => DeploymentManagerDependencies,
   ): IO[ProcessingTypeDataState[ProcessingTypeData, CombinedProcessingTypeData]] = {
-    config
+    processingTypeConfigsLoader
       .loadProcessingTypeConfigs()
-      .map { processingTypesConfig =>
-        // This step with splitting DeploymentManagerProvider loading for all processing types
-        // and after that creating ProcessingTypeData is done because of the deduplication of deployments
-        // See DeploymentManagerProvider.engineSetupIdentity
-        val providerWithNameInputData = processingTypesConfig.mapValuesNow { processingTypeConfig =>
-          val provider = createDeploymentManagerProvider(processingTypeConfig)
-          val nameInputData = EngineNameInputData(
-            provider.defaultEngineSetupName,
-            provider.engineSetupIdentity(processingTypeConfig.deploymentConfig),
-            processingTypeConfig.engineSetupName
-          )
-          (processingTypeConfig, provider, nameInputData)
-        }
-        val engineSetupNames =
-          ScenarioParametersDeterminer.determineEngineSetupNames(providerWithNameInputData.mapValuesNow(_._3))
-        val processingTypesData = providerWithNameInputData
-          .map { case (processingType, (processingTypeConfig, deploymentManagerProvider, _)) =>
-            logger.debug(s"Creating Processing Type: $processingType with config: $processingTypeConfig")
-            val modelDependencies = getModelDependencies(processingType)
-            val processingTypeData = ProcessingTypeData.createProcessingTypeData(
-              processingType,
-              ModelData(processingTypeConfig, modelDependencies),
-              deploymentManagerProvider,
-              getDeploymentManagerDependencies(processingType),
-              engineSetupNames(processingType),
-              processingTypeConfig.deploymentConfig,
-              processingTypeConfig.category,
-              modelDependencies.componentDefinitionExtractionMode
-            )
-            processingType -> processingTypeData
-          }
+      .map(
+        ProcessingTypesConfigBasedProcessingTypeDataLoader
+          .loadProcessingTypeData(_, getModelDependencies, getDeploymentManagerDependencies)
+      )
+  }
+
+}
 
-        // Here all processing types are loaded and we are ready to perform additional configuration validations
-        // to assert the loaded configuration is correct (fail-fast approach).
-        val combinedData = CombinedProcessingTypeData.create(processingTypesData)
+object ProcessingTypesConfigBasedProcessingTypeDataLoader extends LazyLogging {
 
-        ProcessingTypeDataState(
-          processingTypesData.mapValuesNow(toValueWithRestriction),
-          () => combinedData,
-          // We pass here new Object to enforce update of observers
-          new Object
+  def loadProcessingTypeData(
```

Comment:
Hmm for me, if you feel that some code is a nice entrypoint to unit tests, you should prefer to extract this code to some separate, testable method instead of stubbing things that you don't want to test. For example. Let's say that you have a calculator which some GUI for it. In this case you should rather focus on testing logic of functions provided by the calculator instead of ensuring that every test will go through the whole graphical interface. Do you have another opinion on that?

Reply:
as disscussed

Reply:
I replaced this method by passing stub and invocation of unsafeRunSync on result as we discusses